docs/lava-blockchain/join-mainnet-manual-cosmovisor.md

```diff
@@ -0,0 +1,218 @@
+        echo "export PATH=\$PATH:/usr/local/go/bin" >>~/.profile
+        echo "export PATH=\$PATH:\$(go env GOPATH)/bin" >>~/.profile
+        source ~/.profile
+        ```
+        
+    - Installation verifications
+        
+        
+        1. You can verify the installed go version by running: `go version`
+        
+        2. The command `go env GOPATH` should include `$HOME/go`
+        If not, then, `export GOPATH=$HOME/go`
+        
+        3. PATH should include `$HOME/go/bin`
+        To verify PATH, run `echo $PATH`
+        
+
+## 1. Set up a local node
+
+### Download app configurations
+
+- Download setup configuration
+    
+    Download the configuration files needed for the installation
+    
+    ```bash
+    # Download the installation setup configuration
+    git clone https://github.com/lavanet/lava-config.git
+    cd lava-config/mainnet
+    # Read the configuration from the file
+    # Note: you can take a look at the config file and verify configurations
+    source setup_config/setup_config.sh
+    ```
+    
+- Set app configurations
+        
+    Copy lavad default config files to config Lava config folder
+    
+    ```bash
+    echo "Lava config file path: $lava_config_folder"
+    mkdir -p $lavad_home_folder
+    mkdir -p $lava_config_folder
+    cp default_lavad_config_files/* $lava_config_folder
+    ```
+    
+
+### Set the genesis file
+
+- Set the genesis file in the configuration folder
+    
+    ```bash
+    # Copy the genesis.json file to the Lava config folder
+    cp genesis_json/genesis.json $lava_config_folder/genesis.json
+    ```
+
+## 2. Join the Lava Mainnet
+
+The following sections will describe how to install Cosmovisor for automating the upgrades process.
+
+
+### Set up Cosmovisor {#cosmovisor}
+
+- Set up cosmovisor to ensure any future upgrades happen flawlessly. To install Cosmovisor:
+    
+    ```bash
+    go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@v1.0.0
+    # Create the Cosmovisor folder and copy config files to it
+    mkdir -p $lavad_home_folder/cosmovisor/genesis/bin/
+    # Download the genesis binary
+    wget -O  $lavad_home_folder/cosmovisor/genesis/bin/lavad "https://github.com/lavanet/lava/releases/download/v2.2.0/lavad-v2.2.0-linux-amd64"
+    chmod +x $lavad_home_folder/cosmovisor/genesis/bin/lavad
+    ```
+
+    ```bash
+    # Set the environment variables
+    echo "# Setup Cosmovisor" >> ~/.profile
+    echo "export DAEMON_NAME=lavad" >> ~/.profile
+    echo "export CHAIN_ID=lava-mainnet-1" >> ~/.profile
+    echo "export DAEMON_HOME=$HOME/.lava" >> ~/.profile
+    echo "export DAEMON_ALLOW_DOWNLOAD_BINARIES=true" >> ~/.profile
+    echo "export DAEMON_LOG_BUFFER_SIZE=512" >> ~/.profile
+    echo "export DAEMON_RESTART_AFTER_UPGRADE=true" >> ~/.profile
+    echo "export UNSAFE_SKIP_BACKUP=true" >> ~/.profile
+    source ~/.profile
+    ```
+
+    ```bash
+    # Initialize the chain
+    $lavad_home_folder/cosmovisor/genesis/bin/lavad init \
```

Comment:
calling this will overwrite the copied files above

Reply:
We need to perform this initialization to pass the parameters: moniker and chain_id

Reply:
This step is required to specify the Сhain ID and the moniker and this is correct behavior because default values ​​are initially used

Reply:
This step is required to specify the chain ID and moniker and this is correct behavior because default values ​​are initially used

Reply:
this will overwrite the config folder, practically ignoring the setup config and downloading genesis files.
you need to do the "lavad init" before setting up configurations and downloading genesis.
should be in line 90